recipes/capnproto/all/test_package/conanfile.py

```diff
@@ -9,6 +9,11 @@ class TestPackageConan(ConanFile):
     def build_requirements(self):
         if tools.cross_building(self.settings):
             self.build_requires(str(self.requires["capnproto"]))
+        if self.settings.os == "Macos" and self.settings.arch == "armv8":
+            # Attempting to use @rpath without CMAKE_SHARED_LIBRARY_RUNTIME_C_FLAG being
+            # set. This could be because you are using a Mac OS X version less than 10.5
+            # or because CMake's platform configuration is corrupt.
+            self.build_requires("cmake/3.20.1")
```

Comment:
Could this be moved to the main recipe as the fix for the RPATH problems?

Reply:
I found this fix in the test packages for [`boost`](https://github.com/conan-io/conan-center-index/blob/master/recipes/boost/all/test_package/conanfile.py#L11) and [`gsl`](https://github.com/conan-io/conan-center-index/blob/master/recipes/gsl/all/test_package/conanfile.py#L11).

The main recipe builds with autotools on Linux/macOS and doesn't require CMake. It seems like the recent version of CMake is just needed on macOS/M1 to build any package depending on libraries with `@rpath`.

How would it be possible to inject a `build_requires` on the depending package? Is `build_requires` from the main package propagated down?

Reply:
yes, it's cmake bug, and we didn't upgrade cmake on mac slaves (and cannot at the moment, unfortunately)
the workaround should be in test package

Reply:
Please, add `#FIXME` to you comment so we can filter it in the future when looking for technical debt

Reply:
Added the `#FIXME` comment.